src/libbson/src/bson/bson-memory.h

```diff
@@ -58,6 +63,11 @@ BSON_EXPORT (void)
 bson_zero_free (void *mem, size_t size);
 
 
+#define BSON_ALIGNED_ALLOC(T) bson_aligned_alloc (BSON_ALIGNOF (T), sizeof (T))
+#define BSON_ALIGNED_ALLOC0(T) \
+   bson_aligned_alloc0 (BSON_ALIGNOF (T), sizeof (T))
```

Comment:
If you insert an explicit cast-to-`T*`, that can catch incompatible-pointer assignments. i.e. `double* d = BSON_ALIGNED_ALLOC0(int)`